### PR TITLE
ci: add parallelism to reduce tests CI time 

### LINKED ISSUE #2236: Test run time is too long.
**Describe the bug**

1. Test run time is too long. 
1. They are taking 20 minutes to run 
1. This is unacceptable

**To Reproduce**

1. Submit a Pull Request
2. Tests take too long

**Expected behavior**

- Much faster execution

**Actual behavior**

- See above

**Screenshots**

- N/A.

**Additional details**

- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by syedali237**
**What kind of change does this PR introduce?**

Test run time is too long.

**Issue Number:**

Fixes #2236

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**
Screenshot 2024-08-30 at 1 13 04 AM

**If relevant, did you update the documentation?**

N/A

**Summary**

This PR reduces the test run time from 20 minutes to 4 minutes by optimizing the configuration and test cases.

**Does this PR introduce a breaking change?**

N/A

**Other information**

During the commit, a warning was encountered: **DeprecationWarning: The punycode module is deprecated.** This warning caused changes in the following files:

- schema.graphql
- ./src/types/generatedGraphQLTypes.ts
- README.md

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
  - Added a "Videos" section to the contributing guidelines for easier access to helpful resources.
  - Introduced a new field for additional social media URLs in the schema, enhancing user profile options.

- **Bug Fixes**
  - Minor formatting adjustments in the README for improved visual alignment.

- **Chores**
  - Updated test execution strategy for improved performance during testing.
  - Simplified Vitest configuration by removing unnecessary properties and enhancing test execution control.

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Please post videos of the before and after run times.
2. There is no evidence provided that this change will improve performance.

**Comment by syedali237** [comment]
@palisadoes  Below are the videos that demonstrate the difference before and after the changes. Both videos are sped up 4x for easier comparison:

[Before](https://drive.google.com/file/d/1hVQXW3HaZ-Awli2WSCzk8WGelYbgGMBW/view?usp=drive_link)

[After](https://drive.google.com/file/d/1IyuSD9h7_LKTqnnbz4OwJgeKiYFdVr1Z/view?usp=drive_link)

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "prod": "cross-env NODE_ENV=production pm2-runtime start ./build/server.js",
     "start": "cross-env pm2-runtime start ./build/server.js --watch",
     "setup": "tsx setup.ts",
-    "test": "vitest run --pool forks --poolOptions.forks.singleFork --coverage",
+    "test": "vitest run --pool=threads --no-file-parallelism --coverage",
```
Comment (varshith257) [inline package.json:13]: parallelism is fine but may also lead to race conditions and make tests flaky. I suggest to make sure to keep test files separately that depends on the other tests to run in same thread

Comment (varshith257) [reply package.json:13]: If you can create a new ticket for above

Comment (syedali237) [reply package.json:13]: Thanks for the feedback, @varshith257!  🙌
I've created ticket #2491 to track any further issues with parallelism and race conditions for ongoing monitoring and improvements.

**Comment by varshith257** [comment]
I don't know other file changes are necessary here of readme and graphqlgenerated file in this PR. Except this looks good and nice work @syedali237

**Review by palisadoes — APPROVED** [review]

**Pull request merged**